scripts/app.js

```diff
@@ -12,16 +11,42 @@ function blog (data) {
   this.category = data.category;
 }
 
+blog.all = [];
+
 blog.prototype.toHtml = function() {
   var template = Handlebars.compile($('#article_template').text());
   this.daysAgo = parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000);
   return template(this);
 };
 
-blogData.forEach(function(ele) {
-  blogs.push(new blog(ele));
-});
+blog.fetchAll = function () {
+  if (localStorage.blogData) {
+    $.ajax({
+      type: 'HEAD',
+      url: 'data/blogData.json',
+      success: function(data,message,xhr) {
+        var eTag = xhr.getResponseHeader('eTag');
+        if (!localStorage.eTag || eTag !== localStorage.eTag) {
+          localStorage.eTag = eTag;
+        } else {
+          blog.loadAll(JSON.parse(localStorage.blogData));
+        }
+      }
+    });
+    blog.loadAll(JSON.parse(localStorage.blogData));
+    viewer.initIndexPage();
+  } else {
+    $.get('data/blogData.json', function(data){
+      var stringData = JSON.stringify(data);
+      localStorage.setItem('blogData',stringData);
+      blog.loadAll(JSON.parse(localStorage.blogData));
+    });
```

Comment:
This .getJSON request should be the type of thing you are also calling up above where I made the other comment.  Also, since you will now be making the same call in more than one place you should move it out into it's own function so that you can call it in more than one place.